src/extension-consumer/src/produce/mod.rs

```diff
@@ -70,6 +70,15 @@ impl ProduceLogOpt {
         let (cfg, file_records) = self.validate()?;
         let mut producer = fluvio.topic_producer(&cfg.topic).await?;
 
+        // Check if topic exists before reading files or stdin (#697)
```

Comment:
Issue with this approach is that it moves burden of error checking into application.   Instead,  it should return `PartitionNotFound` or `TopicNotFound` from the client.

Reply:
I am unsure if I understand the objection. However, I did just change the error from `IoError::ErrorKind` to `FluvioError::TopicNotFound` to make the errors fit in with the standard.

Reply:
Issue is that we want to make work for any client not just CLI.  @simlay thoughts?

Reply:
Ah, yeah. [`topic_producer`](https://github.com/infinyon/fluvio/blob/dcdfa58286c3472e24a295a1c4c645e2dc8d02c6/src/client/src/client/cluster.rs#L111-L118) should throw this error if the topic doesn't exist.